### PR TITLE
Improve flex list UI

### PR DESCRIPTION
**Pull request by wxiaoguang**
1. There is already `gt-ac`, so no need to introduce `flex-item-center`
2. The `flex-item-baseline` and `.flex-item-icon svg { margin-top: 1px }` seem to be a tricky patch, they don't resolve the root problem, and still cause misalignment in some cases.
    * The root problem is: the "icon" needs to align with the sibling "title"
    * So, make the "icon" and the "title" both have the same height 
3. `flex-text-inline` could only be used if the element is really "inline", otherwise its `vertical-align` would make the box size change. In most cases, `flex-text-block` is good enough.

image

### CODE REVIEW

File: web_src/css/shared/flex-list.css [hunk 1]
```diff
@@ -41,8 +33,10 @@
   color: var(--color-primary) !important;
 }
 
-.flex-item .flex-item-icon svg {
-  margin-top: 1px;
+.flex-item .flex-item-icon {
+  display: flex;
+  align-items: center;
+  height: 20px; /* match the default flex-item-title height*/
```
Comment (silverwind) [inline web_src/css/shared/flex-list.css]:
```suggestion
  height: 20px; /* match the default flex-item-title height */
```

**Review by silverwind — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by wxiaoguang** [comment]
I have to admit that such layout trick (flex "icon" + "title") is not good enough.

For example, if there are labels ....

Maybe the only way for a perfect fix is to get rid of such layout, eg: move the "icon" into the "title", let they layout together.

image

**Comment by wxiaoguang** [comment]
And even worse .... with emoji label:

image

**Comment by silverwind** [comment]
I woul aim to have the baseline of the text aligned with the lower edge of the left side icon. Labels should be centered on text. While doing these tweaks, ensure wrapping does not regress.

**Comment by wxiaoguang** [comment]
-> Improve issue list layout #26983

**Pull request merged**